SafeSnap-Agents/AnswersSubmission/src/agent.spec.ts

```diff
@@ -0,0 +1,211 @@
+import { Interface } from "@ethersproject/abi";
+import {
+  Finding,
+  FindingSeverity,
+  FindingType,
+  HandleTransaction,
+  TransactionEvent,
+} from "forta-agent";
+import { when } from "jest-when";
+import { createAddress, TestTransactionEvent } from "forta-agent-tools";
+import { initialize, provideHandleTransaction } from "./agent";
+import { ORACLE_ABI } from "./utils";
+
+const createFinding = (
+  name: string,
+  questionId: string,
+  answer: string,
+  user: string,
+  commitment: boolean | undefined
+) => {
+  if (name === "LogNewAnswer")
+    return Finding.fromObject({
+      name: "Answers submission alert",
+      description: "An answer has been submitted to a question",
+      alertId: "GNOSIS-3-1",
+      type: FindingType.Info,
+      severity: FindingSeverity.Info,
+      protocol: "Gnosis SafeSnap",
+      metadata: {
+        questionId,
+        answer,
+        user,
+        commitment: commitment ? "yes" : "no",
+      },
+    });
+  else if (name === "LogAnswerReveal")
+    return Finding.fromObject({
+      name: "Answers submission alert",
+      description: "An answer has been revealed",
+      alertId: "GNOSIS-3-2",
+      type: FindingType.Info,
+      severity: FindingSeverity.Info,
+      protocol: "Gnosis SafeSnap",
+      metadata: {
+        questionId,
+        answer,
+        user,
+      },
+    });
+  else
+    return Finding.fromObject({
+      name: "Question finalized alert",
+      description: "A question has been finalized",
+      alertId: "GNOSIS-3-3",
+      type: FindingType.Info,
+      severity: FindingSeverity.Info,
+      protocol: "Gnosis SafeSnap",
+      metadata: {
+        questionId,
+        answer,
+      },
+    });
+};
+
+describe("answers submission agent", () => {
+  const reality_module = createAddress("0xffee");
+  const oracle_address = createAddress("0xfede");
+  const different_oracle = createAddress("0xd4");
+  const WRONG_EVENT_SIGNATURE = "event wrong_sig()";
+  const questions = [
+    // array of test questions.
+    "0x7ca0eb796b737d1d6fbe75cae0c351c4626ce7d826ceff3a9f91ec8d2282cc24",
+    "0x8ca0eb796b737d1d6fbe75cae0c351c4626ce7d826ceff3a9f91ec8d2282cc34",
+    "0x9ca0eb796b737d1d6fbe75cae0c351c4626ce7d826ceff3a9f91ec8d2282cc44",
+    "0xaca0eb796b737d1d6fbe75cae0c351c4626ce7d826ceff3a9f91ec8d2282cc52",
+  ];
+
+  let handler: HandleTransaction;
+  const mockGetOracle = jest.fn();
+  const mockOracleFetcher = {
+    getOracle: mockGetOracle,
+  };
+
+  beforeAll(async () => {
+    //init the mock fetcher
+    when(mockGetOracle)
+      .calledWith("latest", reality_module)
+      .mockReturnValue(oracle_address);
+    // init the agent ie. fetch the oracle.
+    handler = provideHandleTransaction("");
+    const initializer = initialize(reality_module, mockOracleFetcher as any);
+    await initializer();
```

Comment:
You don't need to use the initialize logic here even more when the logic used in the initialize is tested in your fetcher tests. You can keep it that way if you want, but I prefer to see
```
handler = provideHandleTransaction(oracle_address);
```
and that will let you remove the `if(!_oracle) _oracle = oracle` line in your agent logic, that I think is there just for testing purposes.
If you want to keep using the initialize function here I think that you can return the oracle in the function, that is not the expected signature of the initialize function but it will not break anything. By doing that you will be able to get the oracle an create the handler after that.
My preference current preference is to not use the initialize logic in the tests.

Reply:
addressed in the last commit